From 9b593e066635d011272dbe79438dc7508177167f Mon Sep 17 00:00:00 2001
Message-Id: <9b593e066635d011272dbe79438dc7508177167f.1421709247.git.chang-joon.lee@intel.com>
In-Reply-To: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
References: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
From: Shobhit Kumar <shobhit.kumar@intel.com>
Date: Wed, 17 Sep 2014 17:17:56 +0530
Subject: [PATCH 29/75] MUST_REBASE [VPG]: adf/intel: Compile in intel fbdev
 support only if enabled

Keeping by default as disabled because current implementation does not
invoke GEM allocation within KMD for FB allocation for framebuffer
driver.

TBD:
FBDEV driver needs to be enabled

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-3547
Change-Id: Iba0779e574d098e3115e2e6ec25dee860a8e3f4a
Signed-off-by: Shobhit Kumar <shobhit.kumar@intel.com>
---
 drivers/video/adf/intel/intel_adf.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/video/adf/intel/intel_adf.c b/drivers/video/adf/intel/intel_adf.c
index d5a2d60..ef8a6d5 100644
--- a/drivers/video/adf/intel/intel_adf.c
+++ b/drivers/video/adf/intel/intel_adf.c
@@ -111,7 +111,7 @@ static int create_attachments(struct intel_adf_device *dev,
 	return 0;
 }
 
-#if defined(CONFIG_ADF_FBDEV)
+#if defined(CONFIG_ADF_FBDEV) && defined(CONFIG_ADF_INTEL_FBDEV)
 static struct adf_fbdev *create_fbdevs(struct intel_adf_context *ctx)
 {
 	struct intel_adf_device *dev = ctx->dev;
@@ -182,7 +182,7 @@ void intel_adf_context_destroy(struct intel_adf_context *ctx)
 
 	if (!ctx)
 		return;
-#if defined(CONFIG_ADF_FBDEV)
+#if defined(CONFIG_ADF_FBDEV) && defined(CONFIG_ADF_INTEL_FBDEV)
 	if (ctx->fbdevs)
 		destroy_fbdevs(ctx->fbdevs, ctx->n_fbdevs);
 #endif
@@ -212,7 +212,7 @@ struct intel_adf_context *intel_adf_context_create(struct pci_dev *pdev)
 	struct intel_adf_interface *intfs;
 	struct intel_adf_overlay_engine *engs;
 	int n_intfs, n_engs;
-#if defined(CONFIG_ADF_FBDEV)
+#if defined(CONFIG_ADF_FBDEV) && defined(CONFIG_ADF_INTEL_FBDEV)
 	struct adf_fbdev *fbdevs;
 #endif
 	/*TODO: use real platform ID*/
@@ -289,7 +289,7 @@ struct intel_adf_context *intel_adf_context_create(struct pci_dev *pdev)
 
 	g_adf_context = ctx;
 
-#if defined(CONFIG_ADF_FBDEV)
+#if defined(CONFIG_ADF_FBDEV) && defined(CONFIG_ADF_INTEL_FBDEV)
 	fbdevs = create_fbdevs(ctx);
 	if (IS_ERR(fbdevs)) {
 		dev_err(&pdev->dev, "%s: failed to create FB devices\n",
-- 
1.7.9.5

